### ISSUE TITLE
`exa` command (ls equivalent) crashes in neovim terminal (works just in iterm2) Segmentation Fault

### ISSUE THREAD

**Issue by John15321** [neovim/neovim#23987]
### Problem

When I run the exa command with tree flags `exa --tree -L 3 -l --git --icons -F` (which simply function like a tree command) and try to use it to list a big directory. When I do it in just `vim` it doesn't crash.

### Steps to reproduce

Install `exa`, for example using cargo
`cargo install exa`

Got to a sufficiently large directory with many subdirectories and files in them. Run exa with there flags:

```bash
exa --tree -L 3 -l --git --icons -F
```
The `-L 3` is the recursion number for the tree mode, so given my directory structure 3 was enough.
The directory I tried this on is a build directory for my project: https://github.com/PWr-Aerospace/ground-control-station
You can clone it, run the instructions in the readme to build it by running `cargo tauri build`, and then go to `src-tauri/target` and run the beforementioned command there, that would be the exact replication of my problem.

Then simply open neovim in this directory, I did this by changing directory there and running `nvim .` and then `:term`, and inside that terminal running the `exa --tree -L 3 -l --git --icons -F` command. The directory where I noticed it, had 3029 items, but Im guessing less would be enough to cause the failure.

### Expected behavior

Should crash and just print out the items. When I exit neovim and do it using the terminal itself it works, same with the terminal in VSCode

### Neovim version (nvim -v)

NVIM v0.9.0-dev-3299+g266a1c61b-dirty

### Vim (not Nvim) behaves the same?

no, VIM - Vi IMproved 9.0

### Operating system/version

macOS 12.6

### Terminal name/version

iTerm2 3.4.19

### $TERM environment variable

xterm-256color

### Installation

Homebrew

**Comment by clason** [comment]
Please test the latest nightly version.

**Comment by John15321** [comment]
> Please test the latest nightly version.

Can i achive it by running `brew install neovim --HEAD`? @clason

**Comment by clason** [comment]
However you got your last prerelease. 

(brew is one possibility, or you can just download the nightly from our releases page)

**Comment by John15321** [comment]
@clason Im having problems installing the nightly neovim. My os is fairly up to date i would rather not upgrade it but im gettings this error:

```bash
==> Downloading https://formulae.brew.sh/api/formula.jws.json
#=#=-  #       #
==> Downloading https://formulae.brew.sh/api/cask.jws.json
#=#=-  #       #
==> Fetching neovim
==> Downloading https://raw.githubusercontent.com/Homebrew/homebrew-core/aaffc9eafc53704760708c41302018d8b4b1f513/Formula/neovim.rb
Already downloaded: /Users/john/Library/Caches/Homebrew/downloads/f34386c49b35339782d91017c86215609dc7e3ede012b9fe21c6be97d81200e4--neovim.rb
==> Cloning https://github.com/neovim/neovim.git
Updating /Users/john/Library/Caches/Homebrew/neovim--git
==> Checking out branch master
Already on 'master'
Your branch is up to date with 'origin/master'.
HEAD is now at dcb341315 Merge pull request #23968 from figsoda/bundle
Warning: A newer Command Line Tools release is available.
Update them from Software Update in System Preferences.

If that doesn't show you any updates, run:
  sudo rm -rf /Library/Developer/CommandLineTools
  sudo xcode-select --install

Alternatively, manually download them from:
  https://developer.apple.com/download/all/.
You should download the Command Line Tools for Xcode 14.2.

==> cmake -S . -B build -DLUV_LIBRARY=/opt/homebrew/opt/luv/lib/libluv.dylib -DLIBUV_LIBRARY=/opt/homebrew/opt/libuv/lib/libuv.dylib
Last 15 lines from /Users/john/Library/Logs/Homebrew/neovim/01.cmake:
-- Found Iconv: /Library/Developer/CommandLineTools/SDKs/MacOSX12.sdk/usr/include
-- Found Lpeg: /opt/homebrew/lib/lua/5.1/lpeg.so
-- Found Libtermkey: /opt/homebrew/lib/libtermkey.dylib (Required is at least version "0.22")
-- Found Libvterm: /opt/homebrew/include (found suitable version "0.3", minimum required is "0.3")
-- Found Msgpack: /opt/homebrew/lib/libmsgpackc.dylib (found suitable version "6.0.0", minimum required is "1.0.0")
CMake Error at /opt/homebrew/Cellar/cmake/3.26.4/share/cmake/Modules/FindPackageHandleStandardArgs.cmake:230 (message):
  Could NOT find Treesitter (missing: TREESITTER_LIBRARY
  TREESITTER_INCLUDE_DIR) (Required is at least version "0.20.8")
Call Stack (most recent call first):
  /opt/homebrew/Cellar/cmake/3.26.4/share/cmake/Modules/FindPackageHandleStandardArgs.cmake:600 (_FPHSA_FAILURE_MESSAGE)
  cmake/FindTreesitter.cmake:3 (find_package_handle_standard_args)
  src/nvim/CMakeLists.txt:33 (find_package)


-- Configuring incomplete, errors occurred!

READ THIS: https://docs.brew.sh/Troubleshooting


Error: A newer Command Line Tools release is available.
Update them from Software Update in System Preferences.

If that doesn't show you any updates, run:
  sudo rm -rf /Library/Developer/CommandLineTools
  sudo xcode-select --install

Alternatively, manually download them from:
  https://developer.apple.com/download/all/.
You should download the Command Line Tools for Xcode 14.2.

neovim's formula was built from an unstable upstream --HEAD.
This build failure is expected behaviour.
Do not create issues about this on Homebrew's GitHub repositories.
Any opened issues will be immediately closed without response.
Do not ask for help from Homebrew or its maintainers on social media.
You may ask for help in Homebrew's discussions but are unlikely to receive a response.
Try to figure out the problem yourself and submit a fix as a pull request.
We will review it but may or may not accept it.
```

**Comment by zeertzjq** [comment]
The steps to reproduce look a bit complicated. At least, what is the height and width of the terminal window?

**Comment by clason** [comment]
Just follow the instructions (or, as I wrote, download the nightly from the releases page).

**Comment by John15321** [comment]
@zeertzjq @clason Okay, I did what you said, the newest manually downloaded version of nvim (`NVIM v0.10.0-dev-497+g4229bbe51`) does not seem to have this problem

**Comment by John15321** [comment]
Although please dont close this issue yet. I would like to make sure i can install it the "normal" way and then that the same doesnt happen again

**Comment by John15321** [comment]
@clason @zeertzjq I upgraded my os, but im still encountering problems installing it, something is wrong:

```bash
==> Fetching neovim
==> Downloading https://raw.githubusercontent.com/Homebrew/homebrew-core/1e3700662cee0a324b3e0aff9ca2b8419615af01/Formula/neovim.rb
###################################################################################################################################################################################################################################################### 100.0%
==> Cloning https://github.com/neovim/neovim.git
Updating /Users/john/Library/Caches/Homebrew/neovim--git
==> Checking out branch master
Already on 'master'
Your branch is up to date with 'origin/master'.
HEAD is now at bde59e814 fix(remote): restore previous --remote-expr output formatting (#23988)
==> cmake -S . -B build -DLUV_LIBRARY=/opt/homebrew/opt/luv/lib/libluv.dylib -DLIBUV_LIBRARY=/opt/homebrew/opt/libuv/lib/libuv.dylib
Last 15 lines from /Users/john/Library/Logs/Homebrew/neovim/01.cmake:
-- Found Iconv: /Library/Developer/CommandLineTools/SDKs/MacOSX13.sdk/usr/include
-- Found Lpeg: /opt/homebrew/lib/lua/5.1/lpeg.so
-- Found Libtermkey: /opt/homebrew/lib/libtermkey.dylib (Required is at least version "0.22")
-- Found Libvterm: /opt/homebrew/include (found suitable version "0.3", minimum required is "0.3")
-- Found Msgpack: /opt/homebrew/lib/libmsgpackc.dylib (found suitable version "6.0.0", minimum required is "1.0.0")
CMake Error at /opt/homebrew/Cellar/cmake/3.26.4/share/cmake/Modules/FindPackageHandleStandardArgs.cmake:230 (message):
  Could NOT find Treesitter (missing: TREESITTER_LIBRARY
  TREESITTER_INCLUDE_DIR) (Required is at least version "0.20.8")
Call Stack (most recent call first):
  /opt/homebrew/Cellar/cmake/3.26.4/share/cmake/Modules/FindPackageHandleStandardArgs.cmake:600 (_FPHSA_FAILURE_MESSAGE)
  cmake/FindTreesitter.cmake:3 (find_package_handle_standard_args)
  src/nvim/CMakeLists.txt:33 (find_package)


-- Configuring incomplete, errors occurred!

READ THIS: https://docs.brew.sh/Troubleshooting

neovim's formula was built from an unstable upstream --HEAD.
This build failure is expected behaviour.
Do not create issues about this on Homebrew's GitHub repositories.
Any opened issues will be immediately closed without response.
Do not ask for help from Homebrew or its maintainers on social media.
You may ask for help in Homebrew's discussions but are unlikely to receive a response.
Try to figure out the problem yourself and submit a fix as a pull request.
We will review it but may or may not accept it.
```

**Comment by clason** [comment]
I don't know what you're trying to do, but this is clearly a homebrew-specific build failure. Why not just do `brew install neovim` and call it a day?

**Comment by John15321** [comment]
Im running `brew install neovim --HEAD` which simply tries to install it from the latest commit. normal `brew install neovim` installs version 0.9.1

**Comment by clason** [comment]
Yes. I strongly recommend that this is the version you install.

**Comment by John15321** [comment]
Oh, I see, now that im doing it, it works. So it must have been solved in 0.9.0-0.9.1
